packages/dds/tree/src/test/simple-tree/arrayNode.spec.ts

```diff
@@ -721,4 +721,124 @@ describe("ArrayNode", () => {
 			);
 		});
 	});
+
+	describe("Iteration", () => {
+		it("Concurrently iterating and editing should throw an error.", () => {
+			const array = hydrate(CustomizableNumberArray, [1, 2, 3]);
+			const values = array.values();
+			values.next();
+			array.removeRange(1, 3);
+			assert.throws(
+				() => {
+					values.next();
+				},
+				validateUsageError(/Concurrent editing and iteration is not allowed./),
+			);
+			// Checks that new iterator still works
+			const values2 = array.values();
+			values2.next();
+		});
+
+		it("Concurrently iterating and editing after inserting into an unhydrated array node errors.", () => {
```

Comment:
I don't think that either of these "unhydrated" tests are actually testing anything interesting. Once the nodes are hydrated, they're hydrated. Whether they used to be unhydrated, or never existed in an unhydrated state, doesn't matter. 

The interesting test would be to create an iterator for an unhydrated node, _then_ hydrate it, and then check that the iterator keeps working. I expect that this will not work with the `hydrate` function, because our `hydrate` function doesn't actually hydrate unhydrated nodes :) I have a task to fix this but I've been unsuccessful so far.

In the meantime, you can use a real SharedTree from a TestTreeProvider to test this case. I'll go try to get hydration via `hydrate` working in the meantime now that I know we have an actual use for it.

Reply:
Updated test case to:
1. create an unhydrated array node, create an iterator, initialize a tree with the unhydrated node (to hydrate it) and checks that the iterator works
4. make an edit to the node, and checks that it throws a usage error